src/lib/input/input.scss

```diff
@@ -37,6 +37,22 @@
     display: none;
   }
 
+  // Fixes an issue on iOS where the following input types will collapse to 1px,
+  // if they're empty, because we've overridden their background color.
+  // See: https://stackoverflow.com/questions/18381594/input-type-date-appearance-in-safari-on-ios
+  &[type='date'],
+  &[type='datetime'],
+  &[type='datetime-local'],
+  &[type='month'],
+  &[type='week'],
+  &[type='time'] {
+    &::after {
+      content: ' ';
+      white-space: pre;
+      width: 1px;
```

Comment:
did you try adding something like `margin-left: -1px;` just to make extra sure it doesn't mess with the layout, also maybe `visibility: hidden;`

Reply:
I wanted to keep it as minimal as possible. If we added the margin, we'd also have to flip it in RTL. I don't think it's an issue in this case, because users can't type in these input types anyway.